### ISSUE TITLE
rxjs-add false positive when type is not specified explicitly

### ISSUE THREAD

**Issue by balassy** [cartant/rxjs-tslint-rules#13]
I have the following function in an Angular 4 service class:
```
public get(url: string): Promise<Response> {
  return this._http.get(url).toPromise();
}
```
Note, that the `get` function returns an `Observable`, but this type name is not appearing in the code explicitly.
The `.toPromise()` function of `Observable` is imported in `rxjs.imports.ts` like this:
```
import 'rxjs/add/operator/toPromise';
```

And I have the following rule in `tslint.json`:
```
"rxjs-add": {
  "options": [{
      "file": "./src/rxjs.imports.ts"
  }],
  "severity": "error"
}
```
This triggers the following error when running `tslint`, which seems to be a false alarm:
```
ERROR: (rxjs-add) C:/Git/MyProject/src/rxjs.imports.ts[4, 1]: 
Unused patched operator in ./src/rxjs.imports.ts: toPromise
```

Thank you for creating and maintaining this library, Nicholas!

György

**Comment by cartant** [comment]
First question: are you using TSLint's `no-unused-variable` rule? That rule has given me major problems in the past. See #4.

**Comment by balassy** [comment]
Yes, I am using `tslint:all` ruleset, which turns on `no-unused-variable` by default. I've checked now, and turning it off fixes this problem.
Thank you!

**Comment by cartant** [comment]
Yeah, the `no-unused-variable` rule misbehaves badly. It's had problems that have been 'fixed' before, but it's still broken in a way that affects other rules. It's a pity, as it's a useful rule. However, it's wasted so much of my time on previous occasions and is difficult to debug, so I just switch if off.

**Comment by balassy** [comment]
All right, thank you very much!
